bridges/ethereum/src/actors/block_relay_and_poi.rs

```diff
@@ -236,29 +248,49 @@ pub fn block_relay_and_poi(
                                     })
                                     .or_else({
                                         let config = Arc::clone(&config);
-
+                                        let eth_state = Arc::clone(&eth_state);
                                         move |_| {
                                             log::debug!("Trying to relay superblock {:x}", superblock_hash);
-                                            block_relay_contract2
-                                                .call_with_confirmations(
-                                                    "postNewBlock",
-                                                    (superblock_hash, superblock_epoch, dr_merkle_root, tally_merkle_root),
-                                                    eth_account,
-                                                    contract::Options::with(|opt| {
-                                                        opt.gas = config.gas_limits.post_new_block.map(Into::into);
-                                                    }),
-                                                    1,
+                                            if let Some(last_id) = last_id {
```

Comment:
I would add a comment here, something like:

    // Use the same gas price as the last request from this block.
    // If this block does not contain any requests coming from ethereum, default to "None" which will estimate the gas price using the ethereum client.